services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -285,15 +290,14 @@ private boolean isDuplicateKeyError(final Throwable throwable) {
                         Optional.of(device.getVersion())))
                 .recover(error -> {
                     if (isDuplicateKeyError(error)) {
-                        log.debug("Device [{}] already exists for the tenant [{}]", device.getDeviceId(),
-                                device.getTenantId(), error);
+                        log.debug("Device [{}] already exists for the tenant [{}]", device.getDeviceId());
                         TracingHelper.logError(span, String.format("Device [%s] already exists for the tenant [%s]",
                                 device.getDeviceId(), device.getTenantId()));
                         return Future.succeededFuture(
                                 OperationResult.empty(HttpURLConnection.HTTP_CONFLICT));
                     } else {
                         log.error("Error adding device [{}] for the tenant [{}]", device.getDeviceId(),
-                                device.getTenantId(), error);
+                                device.getTenantId());
```

Comment:
why don't you want to log the exception anymore?

Reply:
In the logs, the entire stack traces are getting logged. Since it is also being logged in span, I removed it here. If you think its better to have it there, I could add it again.

Reply:
The tracing need not be always enabled. For better debugging, it is better to have the error there. I will add it again.